src/Tasks.UnitTests/Copy_Tests.cs

```diff
@@ -2052,11 +2051,32 @@ public class CopyHardAndSymbolicLink_Tests
         [Fact]
         public void CopyWithHardAndSymbolicLinks()
         {
+            // Workaround: For some reason when this test runs with all other tests we are getting
+            // the incorrect result from CreateHardLink error message (a message associated with
+            // another test). Calling GetHRForLastWin32Error / GetExceptionForHR seems to clear
+            // out the previous message and allow us to get the right message in the Copy task.
```

Comment:
I'm curious if this is because the MakeHardLink method doesn't ask for the error if we succeeded...perhaps it's being set anyway in some cases? Looking at that method, windows has a different method it calls from mac/linux—they call Marshal.GetLastWin32Error(). Not sure if that's important, but it might explain your comment below.

Reply:
This was in reference to https://github.com/dotnet/msbuild/pull/8211#discussion_r1039581467; you said GetHRForLastWin32Error and GetExceptionForHR cleared the error on windows but not mac/linux; perhaps you have to call GetLastWin32Error?